agent/handlers/agentapi/taskprotection/v1/handlers/handlers_test.go

```diff
@@ -19,26 +19,135 @@ import (
+	testProtectionEnabled      = true
+)
+
+// MockTaskProtectionClientFactory is a mock of TaskProtectionClientFactory interface
+type MockTaskProtectionClientFactory struct {
+	ctrl     *gomock.Controller
+	recorder *MockTaskProtectionClientFactoryMockRecorder
+}
+
+// MockECSClientMockRecorder is the mock recorder for MockECSClient
+type MockTaskProtectionClientFactoryMockRecorder struct {
+	mock *MockTaskProtectionClientFactory
+}
+
+func NewMockTaskProtectionClientFactory(ctrl *gomock.Controller) *MockTaskProtectionClientFactory {
+	mock := &MockTaskProtectionClientFactory{ctrl: ctrl}
+	mock.recorder = &MockTaskProtectionClientFactoryMockRecorder{mock}
+	return mock
+}
+
+func (m *MockTaskProtectionClientFactory) EXPECT() *MockTaskProtectionClientFactoryMockRecorder {
+	return m.recorder
+}
+
+// GetTaskCredentials mocks base method
+func (m *MockTaskProtectionClientFactory) getECSClient(arg0 credentials.Manager, arg1 *task.Task) (api.ECSTaskProtectionSDK, int, error) {
+	m.ctrl.T.Helper()
+	ret := m.ctrl.Call(m, "getECSClient", arg0, arg1)
+	ret0, _ := ret[0].(api.ECSTaskProtectionSDK)
+	ret1, _ := ret[1].(int)
+	ret2, _ := ret[2].(error)
+	return ret0, ret1, ret2
+}
+func (mr *MockTaskProtectionClientFactoryMockRecorder) getECSClient(arg0 interface{}, arg1 *task.Task) *gomock.Call {
+	mr.mock.ctrl.T.Helper()
+	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "getECSClient", reflect.TypeOf((*MockTaskProtectionClientFactory)(nil).getECSClient), arg0, arg1)
+}
+
+// Tests the path for UpdateTaskProtection API
 func TestTaskProtectionPath(t *testing.T) {
 	assert.Equal(t, "/api/{v3EndpointIDMuxName:[^/]*}/task-protection/v1/state", TaskProtectionPath())
 }
 
-// Helper function for running tests for PutTaskProtection handler
-func testPutTaskProtectionHandler(t *testing.T, state dockerstate.TaskEngineState,
-	v3EndpointID string,
+// TestGetECSClientHappyCase tests getECSClient uses credential in credentials manager and
+// returns a ECS client with correct status code and error
+func TestGetECSClientHappyCase(t *testing.T) {
+	testTask := task.Task{
+		Arn:         testTaskArn,
+		ServiceName: testServiceName,
+	}
+	testTask.SetCredentialsID(testTaskCredentialsId)
+
+	testIAMRoleCredentials := credentials.TaskIAMRoleCredentials{
+		IAMRoleCredentials: credentials.IAMRoleCredentials{
+			AccessKeyID:     testAccessKey,
+			SecretAccessKey: testSecretKey,
+			SessionToken:    testSessionToken,
+		},
+	}
+
+	factory := TaskProtectionClientFactory{
+		Region: testRegion, Endpoint: testEndpoint, AcceptInsecureCert: testAcceptInsecureCert,
+	}
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockManager := mock_credentials.NewMockManager(ctrl)
+	mockManager.EXPECT().GetTaskCredentials(gomock.Eq(testTaskCredentialsId)).Return(testIAMRoleCredentials, true)
+
+	ret, statusCode, err := factory.getECSClient(mockManager, &testTask)
+	_, ok := ret.(api.ECSTaskProtectionSDK)
+
+	// Assert response
+	assert.NoError(t, err)
```

Comment:
Should we have this assert before we access `ret.(api.ECSTaskProtectionSDK)`? Also should we use `require.NoError` instead to make the test stop if there was an error. I recently learned that `assert` package functions do not immediately fail the test when an assertion fails.

Reply:
Discussed offline. https://go.dev/tour/methods/15 type assertion is not going to give a panic as long as second value presents.